include/SimCore/KaonPhysics.h

```diff
@@ -0,0 +1,140 @@
+#ifndef SIMCORE_KAON_PHYSICS_H
+#define SIMCORE_KAON_PHYSICS_H
+
+#include <G4DecayTable.hh>
+#include <G4KaonMinus.hh>
+#include <G4KaonPlus.hh>
+#include <G4KaonZeroLong.hh>
+#include <G4KaonZeroShort.hh>
+#include <G4ParticleDefinition.hh>
+#include <G4VDecayChannel.hh>
+#include <G4VPhysicsConstructor.hh>
+#include <numeric>
+#include <vector>
+
+#include "Framework/Configure/Parameters.h"
+#include "Framework/Exception/Exception.h"
+
+namespace simcore {
+
+/**
+ * @class KaonPhysics
+ *
+ * @brief Allows the configuration of properties of kaons
+ * produced in the simulation, in particular setting their lifetime and
+ * branching ratios
+ *
+ */
+
+class KaonPhysics : public G4VPhysicsConstructor {
+ private:
+  /**
+   *
+   * Represents the 6 possible decay channels for charged kaons in Geant4 (See
+   * G4KaonMinus.cc and G4KaonPlus.cc in the Geant4 sources) in the order they
+   * appear in the decay table.
+   *
+   * The processes are
+   * K^+ -> \mu^+ + \nu_\mu
+   * K^+ -> \pi^+ + \pi^0
+   * K^+ -> \pi^+ + \pi^- + \pi^+
+   * K^+ -> \pi^+ + \pi^0 + \pi^0
+   * K^+ -> \pi^0 + e^+ + \nu_e
+   * K^+ -> \pi^0 + \mu^+ + \nu_\mu
+   *
+   * And vice versa for K^-.
+   * The indices here correspond to the position of the branching ratio for
+   * that process in the corresponding parameter as well as the position in
+   * the decay table.
+   */
+  enum ChargedKaonDecayChannel {
+    mu_nu = 0,
+    pi_pi0 = 1,
+    pi_pi_pi = 2,
+    pi_pi0_pi0 = 3,
+    pi0_e_nu = 4,
+    pi0_mu_nu = 5
+  };
+  /*
```

Comment:
I _think_ doxygen _requires_ this to be two `*` but I'm not sure

```suggestion
  /**
```

I'm just adding this here and won't clutter the other stuff with it.

Reply:
Sigh... making comments have semantics, what fun... Especially given that my editor by default doesn't format it like that

Reply:
yep, in order to allow for the possibility of intentionally excluding some comments from the generated docs, you must have [a special comment-block-format](https://www.doxygen.nl/manual/docblocks.html) to tell doxygen to include a comment.